### DISCUSSION TITLE
Automating data loaders?

### DISCUSSION THREAD

**Discussion by LooniesAndSense** [observablehq/framework#1672] [Q&A] [answered]
Hi folks,

Firstly, thanks to everyone who created Observable Framework!

I need to update my public dashboard daily. I receive stock market data from a pricing API, and I wrote loaders for Observable Framework that accommodate that. Everything works fine.

Where I'm a bit stuck is how to deploy this effectively in an automated fashion. Because I have API keys for the market data, I would rather host this myself. Right now, I build manually, push those updates to a Github repo, and then Digital Ocean pick things up from there and deploy on their app platform (automagically).

I could cobble together a shell script via cron (the updates only happen on trading days), but is there a better way to do this?

Thanks!

**Comment by Fil** [comment]
We tend to think that the best way to do this is to use Observable Cloud. See https://observablehq.com/framework/deploying and https://observablehq.com/documentation/data-apps/deploys

**Reply by LooniesAndSense** [reply, marked as answer]
Thanks, @Fil 

I wasn't clear that the build process could run securely in Observable Cloud, but the docs indicate that I can do that.

I have an account now and will play! Thanks!
